src/main/java/joe/commands/Command.java

```diff
@@ -0,0 +1,31 @@
+package joe.commands;
+
+import joe.Storage;
+import joe.TaskList;
+import joe.exceptions.JoeIndexOutOfBoundsException;
+
+/**
+ * An abstract class representing a command that can be executed on the task list.
+ */
+public abstract class Command {
+    protected boolean isExit = false;
+
+    /**
+     * Executes the command.
+     *
+     * @param tasks   The TaskList on which the command should be executed.
+     * @param storage The storage for saving and loading tasks.
+     * @throws JoeIndexOutOfBoundsException If there is an invalid index used in the command.
+     */
+    public abstract String execute(TaskList tasks, Storage storage)
```

Comment:
Consider using a better naming for this function. `execute` sounds slightly vague.